src/psyclone/dynamo0p3.py

```diff
@@ -59,24 +59,45 @@
     FORTRAN_INTENT_NAMES, DataAccess, Literal, Reference, Schedule, \
     CodedKern, Schedule
 
-# First section : Parser specialisations and classes
-
-# Function spaces (FS)
+# --------------------------------------------------------------------------- #
+# ========== First section : Parser specialisations and classes ============= #
+# --------------------------------------------------------------------------- #
+#
+# ---------- Function spaces (FS) ------------------------------------------- #
 # Discontinuous FS
 DISCONTINUOUS_FUNCTION_SPACES = ["w3", "wtheta", "w2v"]
 # Continuous FS
 # Space any_w2 can be w2, w2h or w2v
 CONTINUOUS_FUNCTION_SPACES = ["w0", "w1", "w2", "w2h", "any_w2"]
+
 # Valid FS and FS names
 VALID_FUNCTION_SPACES = DISCONTINUOUS_FUNCTION_SPACES + \
     CONTINUOUS_FUNCTION_SPACES
 
+# Valid any_space metadata (general FS, treated as continuous)
+# TODO (issue #455): add any_space_10
 VALID_ANY_SPACE_NAMES = ["any_space_1", "any_space_2", "any_space_3",
                          "any_space_4", "any_space_5", "any_space_6",
                          "any_space_7", "any_space_8", "any_space_9"]
 
-VALID_FUNCTION_SPACE_NAMES = VALID_FUNCTION_SPACES + VALID_ANY_SPACE_NAMES
+# Valid any_discontinuous_space metadata (general FS known to be discontinuous)
+VALID_ANY_DISCONTINUOUS_SPACE_NAMES = \
+    ["any_discontinuous_space_1", "any_discontinuous_space_2",
+     "any_discontinuous_space_3", "any_discontinuous_space_4",
+     "any_discontinuous_space_5", "any_discontinuous_space_6",
+     "any_discontinuous_space_7", "any_discontinuous_space_8",
+     "any_discontinuous_space_9", "any_discontinuous_space_10"]
+
+# Valid discontinuous FS names (for optimisation purposes)
+VALID_DISCONTINUOUS_FUNCTION_SPACE_NAMES = DISCONTINUOUS_FUNCTION_SPACES + \
```

Comment:
I know you are inheriting this problem, but I don't like the mixing of *FUNCTION_SPACE_NAMES and *FUNCTION_SPACES. IMHO they are all names so should have _NAMES at the end and are all VALID so don't need VALID at the start. Fixing this is outside of the remit of this PR but to make things a little better perhaps VALID_DISCONTINUOUS_FUNCTION_SPACE_NAMES could be called ALL_DISCONTINUOUS_FUNCTION_SPACE_NAMES and VALID_ANY_DISCONTINUOUS_FUNCTION_SPACE_NAMES could be called ANY_DISCONTINUOUS_FUNCTION_SPACE_NAMES?

Reply:
What is the specific remark about this and is there a specific issue to address this? I couldn't find any so I am happy to open a new issue so we don't forget about this.
I'd rather come to an agreement in this issue to not have to modify optimisation scripts again but I am also happy to leave it for another issue as long as there is a comment in the code as a reminder :-)